twill-yarn/src/main/java/org/apache/twill/internal/appmaster/ApplicationMasterService.java

```diff
@@ -222,6 +226,7 @@ protected void doStart() throws Exception {
     // initialize the event handler, if it fails, it will fail the application.
     if (eventHandler != null) {
       eventHandler.initialize(new BasicEventHandlerContext(twillSpec.getEventHandler()));
+      eventHandler.started(twillRuntimeSpec.getTwillAppName(), twillRuntimeSpec.getTwillAppRunId());
```

Comment:
If initialization fails, then the whole app will fail. `TwillController` should be able to catch this